README.md

```diff
@@ -448,4 +448,48 @@ datetime
 2018-04-22 00:00:00+00:00              2
 2018-04-23 00:00:00+00:00              1
 
+```
+
+### Topic search
+
+Returns lists with the mentions of the search phrase grouped by source. The results are in two formats - the messages themselves and the data for building graph representation of the result. For this metric you'll need to set the [api key](#configuration) and have at least 1000 SAN tokens in your account.
+
+Arguments description:
+
+* sources - an array of enum values that specify which sources should be searched. The sources are:
+    1. TELEGRAM
+    2. PROFESSIONAL_TRADERS_CHAT
+    3. REDDIT
+* search_text - a string containing the key words for which the sources should be searched.
+* from_date - a string representation of datetime value according to the iso8601 standard, e.g. "2018-04-16T10:02:19Z"
+* to_date - a string representation of datetime value according to the iso8601 standard, e.g. "2018-04-16T10:02:19Z"
+* interval - an integer followed by one of: `m`, `h`, `d`, `w`
+
+```python
+
+topic_search = san.get(
+    "topic_search",
+    sources="TELEGRAM, PROFESSIONAL_TRADERS_CHAT, REDDIT",
+    search_text="btc moon",
+    from_date="2018-08-01T12:00:00Z",
+    to_date="2018-08-15T12:00:00Z",
+    interval="6h"
+)
+
+```
+
+Example result:
+
+```
+
+                                                                chartsData  \
+professionalTradersChat  [{'mentionsCount': 170, 'datetime': '2018-08-0...
+reddit                   [{'mentionsCount': 316, 'datetime': '2018-08-0...
+telegram                 [{'mentionsCount': 219, 'datetime': '2018-08-0...
+
+                                                                  messages
+professionalTradersChat  [{'text': 'damn if eth does break tendline btc...
+reddit                   [{'text': 'In the last 2 bull markets (April-M...
+telegram                 [{'text': 'Btc dominance increasing', 'datetim...
```

Comment:
I am wondering if we should split these into 2 separate requests, like `topic_search/mentions` and `topic_search/messages`. I am not sure that you need both all the time. Also I think it will be much easier to work with the data if the index is datetime and the columns are `professionalTradersChat`, `reddit` and `telegram` (this is for the mentions).

About the format of the messages: may be again we make an index on datetime with columns `text` and `type` which will be one of `professionalTradersChat`, `reddit` to `telegram`

What do you think?

Reply:
It's not a problem to split this method to two requests for messages and chart data. And I really don't have an opinion about this.

Having an index of the datetime and merging multiple queries is too complicated IMO and I'd prefer to keep it the way we have it now. Anyway, this is going to be a bigger change and if we decide to go that way I won't be able to do it today.